packages/hint/docs/contributor-guide/getting-started/release.md

```diff
@@ -0,0 +1,229 @@
+
+```json
+{
+    ...
+    "dependencies": {
+        ...
+        "@hint/utils-create-server": "^3.2.1"
+        ...
+    },
+    "version": "2.0.1"
+}
+```
+
+After the release script it will look like:
+
+```json
+{
+    ...
+    "dependencies": {
+        ...
+        "@hint/utils-create-server": "^4.0.0"
+        ...
+    },
+    "version": "2.0.2"
+}
+```
+
+This is fine because of the way `semver` works. Version `2.0.1` and prior will download the latest `3.2.x` version of `@utils/create-test-server`. Version `2.0.2` and later will download `4.0.x`.
+
+Now that `@hint/utils-test-server` has been updated, what happens to all the other packages? The dependency will be updated but no version will be published because there will be only one `Upgrade` commit in the history. Those packages have a dependency such as:
+
+```json
+"@hint/utils-test-server": "^2.0.1"
+```
+
+When someone install the package from a clean cache version `2.0.2` will be downloaded among the latest `@hint/utils-create-server"` which is fully compatible with the current code.
+
+### New feature
+
+* `packageA` breaks something → `Breaking: Drastic change in the API`.
+* `packageB` updates the code to adapt to `packageA`'s new API and it can now support a new feature → `New: Shiny feature`.
+
+The release script will:
+* releases `packageA` with a `major` bump: version 2.0.0.
+* add a new commit to `packageB`: `Upgrade: Bump packageA to 2.0.0`. Because the commit history has `New` and `Upgrade`, `pakageB` will be bumped to 1.6.0 (`New` > `Upgrade`).
+
+**⁉️ NOTE**: When looking at the commit history it might be surprising to see something like the following (from newer to older):
+
+```
+Upgrade: Bump packageA to 2.0.0
+New: This shiny new feature
+```
+
+instead of the reverse. Because of the way that monorepos work this is actually expected. Also this will be "hidden" from the changelog as `Upgrade:` and `Chore:` are not outputed in the process.
+
+### Breaking change
+
+* `packageA` breaks something → `Breaking: Drastic change in the API`.
+* `packageB` changes also needs to change completely to make sure how everything works → `Breaking: Drastic change in the API`.
+
+The release script will:
+* releases `packageA` with a `major` bump: version 2.0.0.
+* add a new commit to `packageB`: `Upgrade: Bump packageA to 2.0.0`. Because the commit history has `Breaking` and `Upgrade`, `packageB` will be bumped to 2.0.0 (`Breaking` > `Upgrade`).
+
+E.g.: If `hint` changes the expectations of the resources should look like, updating will be a `Breaking`.
+
+**⁉️ NOTE**: `packageA` commit could be anything, the important part is how the changes are handled in `packageB` and how they are commited. It is very tempting to do all the changes in one single commit but this could have very bad consequences (e.g.: bumping to a `major` all packages when it should be a `patch`).
+
```

Comment:
Suggested: (e.g.: bumping all packages by a `major` version when it should be a `patch`).